decoder.py

```diff
@@ -118,14 +119,30 @@ def forward(self, input, encoder_outs, hidden_init, targets_len):
             # print(f'log_prob shape {log_prob.shape}')
             # print(f'log_prob {log_prob}')
 
+            # FOR UNK POSTPROCESSING
+            # first, get the src token index with the max attention for this time step
+            src_token_idxs_t = torch.argmax(attn_scores, dim=1)
+            # print(f'src_token_idxs_t shape {src_token_idxs_t.shape}')
+
+            # convert the index location in the src to the vocab class label for src
+            src_labels_t = []
+            for sen_idx in range(len(src_token_idxs_t)):
+                class_label = answer[sen_idx][src_token_idxs_t[sen_idx]]
```

Comment:
when you implement this outside of decoder in train loop, you can potentially use source_vocab instead